Exercise_6/ReversePolishNotation.js

```diff
@@ -0,0 +1,44 @@
+// const process = require('process');
+// console.log(calculate(process.argv[2]));
+
+function calculate(argument,extraOperators) {
+    const invalidArgumentError = new Error('The argument ' + argument + ' is invalid');
+    const signs = argument.split(' ');
+    const stack=[];
+
+    function isNumber(sign) {
+        return !isNaN(parseInt(sign)) && isFinite(sign);
+    }
+
+    function isValidOperator(sign) {
+        const isValidOperatorRegex = new RegExp(/^[+\-*]$/);
+        return isValidOperatorRegex.test(sign);
+    }
+
+
+    signs.reduce((stack,sign) => {
+        if (isNumber(sign)) {
+            stack.push(parseInt(sign));
+        } else if (isValidOperator(sign)) {
+            if (stack.length < 2) throw invalidArgumentError;
+            const firstValue = stack.pop();
+            const secondValue = stack.pop();
+            switch (sign) {
```

Comment:
I don't like the idea of using "switch". It is better to use a hash map. You should create an object which should have a mapping like this: `"+": (first, second) => (second + first)`. Then you will not need a RegExp to check a sign. You will only need to check if the hash map has the key to execute. And no need in "switch".